ProcSDF/GUI/Nodes/CustomNode.cpp

```diff
@@ -0,0 +1,89 @@
+#include "CustomNode.h"
+#include "Common/os.h"
+#include "GUI/NodeGraph.h"
+#include <iostream>
+#include <regex>
+#include <string>
+#include <vector>
+std::vector<std::string> CustomNode::tokenizeWithDelimiters(std::string p_string, std::string p_delimiter)
+{
+	std::regex l_reg(p_delimiter);
+
+	std::sregex_token_iterator l_iter(p_string.begin(), p_string.end(), l_reg, -1);
+	std::sregex_token_iterator l_end;
+
+	std::vector<std::string> l_tokens(l_iter, l_end);
+
+	return l_tokens;
+}
+
+CustomNode::CustomNode(std::string p_filePath)
+{
+	std::string l_fileContent = OS::fetchFileContent(p_filePath);
+	std::vector<std::string> l_lines = tokenizeWithDelimiters(l_fileContent, "\n");
+	for (std::string line : l_lines)
+	{
+		std::vector<std::string> l_tokens = tokenizeWithDelimiters(line, "\\s+");
+
+		if (l_tokens[0] == "//")
```

Comment:
this way of passing metadata is not parsing-friendly and results in ugly code, we need to change this in the future. It's okay in this pr, though.

Reply:
explain? we could just get it done with in this PR..

Reply:
For example, we could use JSON for metadata passing, libraries like jsoncpp can reduce the logic to < 10 lines of code. That would require additional work of integrating a library. Your call if you want to do it in this pr.

Reply:
Nevermind, merge